### DISCUSSION TITLE
de(bi/vu)an bookworm install, php-mapi not available

### DISCUSSION THREAD

**Discussion by cafn** [Z-Hub/Z-Push#42] [Q&A] [answered]
I am trying to install z-push v.2.7.1 onto a de(bi/vu)an bookworm release and when accessing the /Microsoft-Server-ActiveSync url I get a message saying that the module php-mapi is not available.  It is not installed as it is not in the debian repositories.  The installation instructions on [this page](https://github.com/Z-Hub/Z-Push/wiki/Installation#debian-und-ubuntu-based-systems) does not list it as a dependency for debian based systems.  The mail server is not online, yet, as I am still testing.

> FatalMisconfigurationException
> PHP-MAPI extension is not available

Is there a configuation oversight that I have missed somewhere?  Any other ideas?

This is the extended error page that is viewed:

> Trace:
> Array
> (
>     [0] => Array
>         (
>             [file] => /usr/share/z-push/backend/kopano/kopano.php
>             [line] => 41
>             [function] => include_once
>         )
> 
>     [1] => Array
>         (
>             [file] => /usr/share/z-push/vendor/composer/ClassLoader.php
>             [line] => 444
>             [args] => Array
>                 (
>                     [0] => /usr/share/z-push/backend/kopano/kopano.php
>                 )
> 
>             [function] => include
>         )
> 
>     [2] => Array
>         (
>             [file] => /usr/share/z-push/vendor/composer/ClassLoader.php
>             [line] => 322
>             [function] => Composer\Autoload\includeFile
>         )
> 
>     [3] => Array
>         (
>             [function] => loadClass
>             [class] => Composer\Autoload\ClassLoader
>             [type] => ->
>         )
> 
>     [4] => Array
>         (
>             [file] => /usr/share/z-push/lib/core/zpush.php
>             [line] => 627
>             [function] => class_exists
>         )
> 
>     [5] => Array
>         (
>             [file] => /usr/share/z-push/lib/core/zpush.php
>             [line] => 462
>             [function] => GetBackend
>             [class] => ZPush
>             [type] => ::
>         )
> 
>     [6] => Array
>         (
>             [file] => /usr/share/z-push/lib/core/zpush.php
>             [line] => 447
>             [function] => GetStateMachine
>             [class] => ZPush
>             [type] => ::
>         )
> 
>     [7] => Array
>         (
>             [file] => /usr/share/z-push/index.php
>             [line] => 58
>             [function] => CheckAdvancedConfig
>             [class] => ZPush
>             [type] => ::
>         )
> 
> )

**Comment by matidau** [comment]
The section for the installation page you linked to is for before 2.7.0, this was copied directly from the Kopano managed wiki. This is for historical info, and will need to be cleaned up at some point.

Php-mapi is only a requirement for running the Kopano Backend. Are you planning on running Kopano as your mail server? 

It will be a little hard to set up and test Z-Push before setting up the mail server.

**Reply by cafn** [reply]
I will have to research this regarding the Kopano Backend.  

I successfully configured the debian v.11 packages a couple years ago and it has worked flawless.  I am mostly using those configs to apply to this install.  I will have the server in place and plugged in soon and will give it a more thorough testing.  Thank you very much for responding.

Thank you devs for continuing to develop Z-Push!

**Reply by matidau** [reply, marked as answer]
I asked after what mail server you intended to use, as it has come up before where people that intended to use IMAP or Zimbra but had left the default backend for Kopano in the configuration. Sorry if that caused confusion.

The .deb files for php7-mapi and it's dependencies can be obtained from https://download.kopano.io/community/core%3A/
You should use the same .deb packages (or source) as your Kopano server.

Would be keen to know if you successfully get it working with Debian Bookworm.

> Thank you devs for continuing to develop Z-Push!

No Problems!

Cheers,
Mat

**Reply by cafn** [reply]
Actually, I am using the guide located here:  https://workaround.org/ispmail-bookworm/  to build my mail server.  So it is an IMAP mail server.
I have successfully used the previous version of Z-Push (2.4.5-2) using the packages from the debian repository with the the previous guide:  https://workaround.org/ispmail-bullseye/ and, like I said, it has worked absolutely flawless for over two years.

I am just a beginner at understanding how Z-Push works, but I know that it can work well.  I read where one way to tell if z-push is working is to try and access the site at the /Microsoft-Server-ActiveSync and note any errors.  Well I tried it on the new build and received the quoted error response about php-mapi not being available.  Well, I also tried on the mail server currently in use and got the same error response, and it is working fine.  So this may not be a "fatal" error, but just a misleading warning.

Thank you for the *.deb package information.  That might come in handy soon.

**Reply by matidau** [reply]
The Php-mapi error means you have it configured for the Kopano Backend not IMAP.

If you are building an Imap server, I would recommend that you install with MiaB https://github.com/mail-in-a-box/mailinabox  as this has Z-Push configured out of the box and is great as a starting point.

If you want to do your own Imap set up, review the zpush.sh from the MiaB project as this has most of the configuration needed. The only other part is the script to setup nginx.

Cheers,
Mat

**Reply by cafn** [reply]
Thank you very much!  I will look at that zpush.sh file.  However, I chose to not use MaiB as I prefer to not use "one-click appliance installs" and systemd.  This is my personal choice - my build, my responsibility.

Thanks again for you input and assistance!

**Reply by matidau** [reply]
> Thanks again for you input and assistance!

No problems 😊
